akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -382,22 +380,21 @@ case class PhasedFusingActorMaterializer(
       subflowFuser,
       initialAttributes,
       PhasedFusingActorMaterializer.DefaultPhase,
-      PhasedFusingActorMaterializer.DefaultPhases
-    )
+      PhasedFusingActorMaterializer.DefaultPhases)
   }
 
   def materialize[Mat](
     graph:             Graph[ClosedShape, Mat],
     subflowFuser:      GraphInterpreterShell ⇒ ActorRef,
     initialAttributes: Attributes,
     defaultPhase:      Phase[Any],
-    phases:            Map[IslandTag, Phase[Any]]
-  ): Mat = {
-    val islandTracking = new IslandTracking(phases, settings, defaultPhase, this)
+    phases:            Map[IslandTag, Phase[Any]]): Mat = {
+    val islandTracking = new IslandTracking(phases, settings, defaultPhase, this, stageNamePrefix = createFlowName() + "-")
 
     var current: Traversal = graph.traversalBuilder.traversal
 
     val attributesStack = new java.util.ArrayDeque[Attributes](8)
+    println(s"# initialAttributes $initialAttributes") // FIXME
     attributesStack.addLast(initialAttributes)
```

Comment:
Something is different between the `FileSource` and the `FileSink`. The default attributes are lost for the `FileSink`. I can't see any difference in how those are created, both use attributes with `IODispatcher`.

FileSource, see push attr:
```
--- Materializing layout:
 | enter island akka.stream.impl.SinkModuleIslandTag$@7a3653bf
 |  | materialize StreamTestKit$ProbeSink [1a3cde03] []
 | push attr Attributes(List(Name(fileSource), Dispatcher(akka.stream.default-blocking-io-dispatcher), Name(my-fileSource)))
 |  | enter island akka.stream.impl.SourceModuleIslandTag$@428f0266
 |  |  | materialize FileSource(/file-source-spec-lines_22536331993622738109516tmp, 8192) [453499c5] [-1]
 | pop attr
pop mat
```

FileSink, see push attr:
```
--- Materializing layout:
 | push attr Attributes(List(Name(my-fileSink)))
 |  | enter island akka.stream.impl.SinkModuleIslandTag$@25c5089d
 |  |  | materialize FileSink(/synchronous-file-sink6130929689581597644.tmp, Set(WRITE, CREATE)) [0b4474f5] []
 | pop attr
 | push attr Attributes(List(Name(iterableSource)))
 |  |  | push attr Attributes(List())
 |  |  | materialize GraphStage(StatefulMapConcat) [4bf87f4b] [-1]
 |  |  | pop attr
 |  |  | pop mat
 |  | materialize GraphStage(SingleSource(() => Iterator)) [7d85d4b6] [-1]
 | pop attr
pop mat
```